src/benchmark/operators/join_aggregate_benchmark.cpp

```diff
@@ -0,0 +1,181 @@
+        ids_vector.begin() + (chunk_index * chunk_size), ids_vector.begin() + ((chunk_index + 1) * chunk_size)));
+    const auto value_segment = std::make_shared<ValueSegment<int32_t>>(pmr_vector<int32_t>(
+        values.begin() + (chunk_index * chunk_size), values.begin() + ((chunk_index + 1) * chunk_size)));
+    Segments segments;
+    segments.push_back(ids_value_segment);
+    segments.push_back(value_segment);
+
+    table->append_chunk({segments});
+  }
+
+  return table;
+}
+
+std::shared_ptr<TableWrapper> create_zip_table(const size_t table_size) {
+  auto zip_values = generate_zip_codes(table_size);
+
+  auto zip_table = create_table(table_size, zip_values);
+
+  const auto chunk_count = zip_table->chunk_count();
+  for (ChunkID chunk_index = ChunkID{0}; chunk_index < chunk_count; ++chunk_index) {
+    auto chunk = zip_table->get_chunk(chunk_index);
+    chunk->finalize();
+    chunk->set_ordered_by(std::make_pair(ColumnID{0}, OrderByMode::Ascending));
+    chunk->set_ordered_by(std::make_pair(ColumnID{1}, OrderByMode::Ascending));
+  }
+
+  return std::make_shared<TableWrapper>(zip_table);
+}
+
+std::shared_ptr<TableWrapper> create_ages_table(const size_t table_size) {
+  auto ages_values = generate_ages(table_size);
+
+  auto ages_table = create_table(table_size, ages_values);
+
+  const auto chunk_count = ages_table->chunk_count();
+  for (ChunkID chunk_index = ChunkID{0}; chunk_index < chunk_count; ++chunk_index) {
+    auto chunk = ages_table->get_chunk(chunk_index);
+    chunk->finalize();
+    chunk->set_ordered_by(std::make_pair(ColumnID{0}, OrderByMode::Ascending));
+  }
+
+  return std::make_shared<TableWrapper>(ages_table);
+}
+
+template <typename AggType, typename JoinType>
+void BM_Join_Aggregate(benchmark::State& state) {
+  auto table_wrapper_left = create_ages_table(TABLE_SIZE);
+  table_wrapper_left->execute();
+  auto table_wrapper_right = create_zip_table(TABLE_SIZE);
+  table_wrapper_right->execute();
+
+  auto operator_join_predicate =
+      OperatorJoinPredicate(std::make_pair(ColumnID{0}, ColumnID{0}), PredicateCondition::Equals);
+
+  auto aggregates = std::vector<std::shared_ptr<AggregateExpression>>{
+      std::static_pointer_cast<AggregateExpression>(avg_(pqp_column_(ColumnID{0}, DataType::Int, false, "b")))};
+
+  std::vector<ColumnID> groupby = {ColumnID{0}, ColumnID{2}};
+
+  auto warmup_join =
+      std::make_shared<JoinType>(table_wrapper_left, table_wrapper_right, JoinMode::Inner, operator_join_predicate);
+  warmup_join->execute();
+
+  auto warmup_aggregate_sort = std::make_shared<AggType>(warmup_join, aggregates, groupby);
+  warmup_aggregate_sort->execute();
+
+  for (auto _ : state) {
+    auto join =
+        std::make_shared<JoinType>(table_wrapper_left, table_wrapper_right, JoinMode::Inner, operator_join_predicate);
+    join->execute();
+    auto aggregate_sort = std::make_shared<AggType>(join, aggregates, groupby);
```

Comment:
Change to `warmup_aggregate` and `aggregate` since `AggType` describes the type of aggregate operator to be used (either `AggregateSort` or `AggregateHash`)

Reply:
I was actually talking about AggType, which is unnecessarily abbreviated.

Reply:
Aaah :D Sorry